### PR TITLE
Add terratest to test the KMS module.

### PR DESCRIPTION
**Pull request by justin-o12**
This adds a simple terratest test that compares the expected and actual alias. 

Tested with `go test -v` in the `test` dir.

### CODE REVIEW

File: test/example_complete_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package test
+
+import (
+  "testing"
+
+  "github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
```
Comment (justin-o12) [inline test/example_complete_test.go:7]: ugh a tab snuck in there

**Review by timblaktu — COMMENTED** [review]
@justin-o12 LGTM, but I would just ask, what is the main purpose here, are you just trying to get a simple terra-smoke-test in place as a proof of concept, or is this something more?

If the latter, I think we'd want to have more test cases that run against the same test "fixture" created by `InitAndApply`.

Regardless of whether we've got 1 or 100 test cases to run, I generally like to see the assertions in tests happening inside appropriately-named functions. So, at the end of `TestExampleComplete`, you'd call a new function called something like `validateKMSKeyAlias` which runs those last 2 statements. This way, test results/reports will indicate much more precisely what went wrong, just by a more coherent stack trace. As it is now, the assertion would happen in something called `TestExampleComplete`.. not very descriptive. 

I'm probably stating the obvious. Let's check in tomorrow and I'm happy to approve if the intent was to just validate the key alias.

**Comment by justin-o12** [comment]
> @justin-o12 LGTM, but I would just ask, what is the main purpose here, are you just trying to get a simple terra-smoke-test in place as a proof of concept, or is this something more?
> 
> If the latter, I think we'd want to have more test cases that run against the same test "fixture" created by `InitAndApply`.
> 
> Regardless of whether we've got 1 or 100 test cases to run, I generally like to see the assertions in tests happening inside appropriately-named functions. So, at the end of `TestExampleComplete`, you'd call a new function called something like `validateKMSKeyAlias` which runs those last 2 statements. This way, test results/reports will indicate much more precisely what went wrong, just by a more coherent stack trace. As it is now, the assertion would happen in something called `TestExampleComplete`.. not very descriptive.
> 
> I'm probably stating the obvious. Let's check in tomorrow and I'm happy to approve if the intent was to just validate the key alias.

Thanks for the feedback, @timblaktu.  That's a great point about the descriptive stack trace and function name.  I can make that change to make it more helpful.

Regarding a smoke test vs something more:  We can do more here if necessary.  My thinking is this is a pretty simple module and there is little to test here.  I'm definitely open to suggestions on a more helpful test case.

**Comment by zachariahmiller** [comment]
@timblaktu I would welcome you to open a PR to add an additional example (say s3 bucket with kms key) and a test for that example that checks that the bucket is there utilizing the key and functions or something like that. Good feedback, but baby steps. This test would be good in both the s3 and kms modules. Since we will be using multiple tests this process in general will need to be refactored a bit to support that (main_test.go test entry point in the test directory)

**Review by timblaktu — APPROVED** [review]

**Review by wirewc — APPROVED** [review]
LGTM.

**Pull request merged**